packages/runtime/id-compressor/src/idCompressor.ts

```diff
@@ -227,6 +238,40 @@ export class IdCompressor implements IIdCompressor, IIdCompressorCore {
 		return range;
 	}
 
+	public retakeOutstandingCreationRange(): IdCreationRange {
```

Comment:
I generally agree, and I do wonder if a just-in-time approach to sending ranges (like, as we flush the batch manager) would result in us only ever needing this new method. I also like the name getUnfinalizedCreationRange and will probably change it.

Reply:
Sadly I had to make the method mutating (it has take semantics now, since I realized the ordering of generate, takeNext, generate, retakeOutstanding, takeNext relies on mutation in retaking if we want the last takeNext to return an empty range...if it doesn't, we get overlapping range bugs). So, the name stays :(

Reply:
I went with takeUnfinalizedCreationRange :)